### PR TITLE
[REG-1335] Docs for RGGizmos

### PR DESCRIPTION
**Pull request by vontell**
Adds docs for the new RGGizmos debug features in the Unity SDK. These only apply to C# bots, and as such this is included in the C# reference section.

Loom video for the rendering in case you don't want to pull it yourself:

<div>
    <a href="https://www.loom.com/share/66f5e2f994a44e3996e7f21b27206ea5">
      <p>Utilities for C# Bots | RGGizmos Docs - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/66f5e2f994a44e3996e7f21b27206ea5">
      
    </a>
  </div>

### CODE REVIEW

File: docs/studios/unity/unity-sdk/csharp/utilities.mdx [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+
+## RGGizmos
+
+`RGGizmos` is a utility for drawing specific Gizmos and debug information within your Unity scene. It is
+primarily a wrapper around Unity's built-in [Gizmos feature](https://docs.unity3d.com/ScriptReference/Gizmos.html),
+but it also provides some additional functionality around text drawing and rendering Gizmos from the bot threads.
+The RGGizmos library currently supports three main features:
+
+* Drawing lines betweens entities and positions
+* Drawing speheres at entities and positions
```
Comment (addisonbgross) [inline docs/studios/unity/unity-sdk/csharp/utilities.mdx:18]:
```suggestion
* Drawing spheres at entities and positions
```

File: docs/studios/unity/unity-sdk/csharp/utilities.mdx [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+:::caution
+
+Because these billboard text objects are not using the Unity Gizmos framework, they will still be present in the scene
+if you create them and then turn off Gizmos. If you want to remove them, you must call `DestroyAllTexts()`.
+
+:::
+
+### Putting it all together
+
+Let's put this all together! Here is a snippet of code from the a bot written for Unity's Kart Racer sample. This demonstrates
```
Comment (addisonbgross) [inline docs/studios/unity/unity-sdk/csharp/utilities.mdx:276]: Should we add a link to the Kart Racer sample?

https://learn.unity.com/project/karting-template

**Review by addisonbgross — COMMENTED** [review]
Nothing big. Looks good overall

File: docs/studios/unity/unity-sdk/csharp/utilities.mdx [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+but it also provides some additional functionality around text drawing and rendering Gizmos from the bot threads.
+The RGGizmos library currently supports three main features:
+
+* Drawing lines betweens entities and positions
+* Drawing speheres at entities and positions
+* Rendering billboard text above an entity 
+
+**Skip to the end of the RGGizmos section to see a video of these in action.**
+
+All methods below can be accessed via the `RGGizmos` variable in your `ProcessTick(RG rgObject)` method.
```
Comment (RG-nAmKcAz) [inline docs/studios/unity/unity-sdk/csharp/utilities.mdx:23]: I feel like this sentence needs some wording qualification that `ProcessTick` is within your bot code.

**Review by RG-nAmKcAz — APPROVED** [review]

File: docs/studios/unity/unity-sdk/csharp/utilities.mdx (hunk 3 above)
Comment (vontell) [reply docs/studios/unity/unity-sdk/csharp/utilities.mdx:23]: Fixed! Also linked to our only page that talks about ProcessTick until we add a proper reference.

File: docs/studios/unity/unity-sdk/csharp/utilities.mdx (hunk 2 above)
Comment (vontell) [reply docs/studios/unity/unity-sdk/csharp/utilities.mdx:276]: Fixed!

File: docs/studios/unity/unity-sdk/csharp/utilities.mdx (hunk 1 above)
Comment (vontell) [reply docs/studios/unity/unity-sdk/csharp/utilities.mdx:18]: Fixed!

**Review by addisonbgross — APPROVED** [review]

**Review by abeizer — APPROVED** [review]

**Review by RG-nAmKcAz — APPROVED** [review]

**Pull request merged**